src/FilesystemManagement.cpp

```diff
@@ -0,0 +1,365 @@
+// � Copyright 2010 - 2019 BlackTopp Studios Inc.
+/* This file is part of The Mezzanine Engine.
+
+    The Mezzanine Engine is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The Mezzanine Engine is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with The Mezzanine Engine.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/* The original authors have included a copy of the license specified above in the
+   'Docs' folder. See 'gpl.txt'
+*/
+/* We welcome the use of the Mezzanine engine to anyone, including companies who wish to
+   Build professional software and charge for their product.
+
+   However there are some practical restrictions, so if your project involves
+   any of the following you should contact us and we will try to work something
+   out:
+    - DRM or Copy Protection of any kind(except Copyrights)
+    - Software Patents You Do Not Wish to Freely License
+    - Any Kind of Linking to Non-GPL licensed Works
+    - Are Currently In Violation of Another Copyright Holder's GPL License
+    - If You want to change our code and not add a few hundred MB of stuff to
+        your distribution
+
+   These and other limitations could cause serious legal problems if you ignore
+   them, so it is best to simply contact us or the Free Software Foundation, if
+   you have any questions.
+
+   Joseph Toppi - toppij@gmail.com
+   John Blackwood - makoenergy02@gmail.com
+*/
+
+#include "CrossPlatformExport.h"
+
+#ifdef MEZZ_Windows
+    #define _WIN32_WINNT 0x0601
+#endif
+
+#include "FilesystemManagement.h"
+#include "PathUtilities.h"
+#include "StringTools.h"
+
+#include <cstring>
+#include <iostream>
+
+#ifdef MEZZ_Windows
+    #define WIN32_LEAN_AND_MEAN
+
+    SAVE_WARNING_STATE
+    SUPPRESS_VC_WARNING(4668)
+
+    #include <Windows.h>
+
+    RESTORE_WARNING_STATE
+#else
+    #include <stdio.h>
+    #include <sys/stat.h>
+    #include <sys/types.h>
+    #include <unistd.h>
+#endif
+
+#include "PlatformUndefs.h"
+
+namespace
+{
+    using namespace Mezzanine;
+
+#ifdef MEZZ_Windows
+    /// @brief Converts a narrow (8-bit) string to a wide (16-bit) string.
+    /// @param Narrow The string to be converted.
+    /// @return Returns a wide string with the converted contents.
+    std::wstring ConvertToWideString(const StringView Narrow)
+    {
+        std::wstring Ret;
+        if( !Narrow.empty() ) {
+            int NarrowSize = static_cast<int>(Narrow.size());
```

Comment:
What about strings that are 2~4 GB in size? Shouldn't there be a check against `std::numeric_limits<int>::max()`?

Reply:
Added.